opsdroid/connector/gitter/__init__.py

```diff
@@ -85,12 +107,12 @@ async def parse_message(self, message):
                 return Message(
                     text=message["text"],
                     user=message["fromUser"]["username"],
+                    user_id=message["fromUser"]["id"],
                     target=self.room_id,
                     connector=self,
                 )
-            except KeyError as err:
-                _LOGGER.error(_("Unable to parse message %s."), err)
-                _LOGGER.error(err)
+            except KeyError:
+                _LOGGER.exception(_("Unable to parse message %r."), message)
```

Comment:
P.S. If an `exception` log (which implicitly includes the traceback of the current exception) is considered too verbose, I'd propose an `error` log that includes the _repr_ of the exception (e.g. `KeyError("text")`) and the content of `message`.

Reply:
We try to avoid leaking message text into the logs. The bot may be able to access rooms that folks with log reading privileges shouldn't have access to. This matters more for chat services with private rooms.

Reply:
Oh, that makes sense. Force-pushed out this commit, replaced it a different approach closer to the original but a little more helpful.